templates/apilb.conf

```
@@ -32,7 +32,7 @@ server {
       proxy_set_header        X-Forwarded-Proto-Version $http2;
       proxy_set_header        Upgrade $http_upgrade;
       proxy_set_header        Connection $http_connection;
-      proxy_set_header        X-Stream-Protocol-Version $http_x_stream_protocol_version;
+      proxy_set_header        X-Stream-Protocol-Version "{{ x_stream_protocol_version }}";
```

if the `x_stream_protocol_version` is unset on the charm, we should likely go back to the default behavior

```suggestion
      proxy_set_header        X-Stream-Protocol-Version "{{ x_stream_protocol_version or '$http_x_stream_protocol_version' }}";
```

---

hi @addyess , Using an empty value will cause an error 'Error from server:', pls see my test result[1] for more details.
and the previous default value was also not empty. nginx does not have the ability to merge multiple headers, so the default value of http_x_stream_protocol_version should be v5.channel.k8s.io or v4.channel.k8s.io by $http_x_stream_protocol_version before.

I also did another test[2] by setting this configuration back to the previous vaule of '$http_x_stream_protocol_version', and the problem happened again.

Therefore, I believe setting 'v5.channel.k8s.io, v4.channel.k8s.io, v3.channel.k8s.io, v2.channel.k8s.io, channel.k8s.io' as default value is appropriate. The current issue is what vaule should be set when we reset it, setting it to empty will cause above error 'Error from server'. To address this, I will reset it to the previous  value of '$http_x_stream_protocol_version'. This way, I will not change the original behavior of this charm, allowing customers who need it to set this parameter as well. If you have any better suggestions, please let me know. Thank you

[1] https://pastebin.canonical.com/p/HxMGVXksR7/
[2] https://pastebin.canonical.com/p/Wbj73NDgZc/